hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/NextKSketch.java

```diff
@@ -0,0 +1,153 @@
+package org.hiero.sketch.spreadsheet;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.hiero.sketch.dataset.api.ISketch;
+import org.hiero.sketch.dataset.api.PartialResult;
+import org.hiero.sketch.table.*;
+import org.hiero.sketch.table.api.*;
+import rx.Observable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedMap;
+
+public class NextKSketch implements ISketch<ITable, NextKList> {
+    @NonNull
+    private final RecordOrder recordOrder;
+    private final RowSnapshot topRow;
+    private final int maxSize;
+
+    public NextKSketch(@NonNull RecordOrder recordOrder, @NonNull RowSnapshot topRow, int maxSize) {
```

Comment:
Actually RowSnapshot can be null - if we want to start from the first row.
The JavaDoc comment for the constructor should specify this fact.